### ISSUE TITLE
Jam not working: 502 Gateway Error

### ISSUE THREAD

**Issue by nobodyhere0** [joinmarket-webui/jam#871]
I'm running a umbrel server on a PI 4 using umbrelOS 1.2.2.  Bitcoin Node v28 is also installed and working.  Jam 0.3.0 is installed but for some reason it no longer works.  It was running fine 3 or 4 weeks ago without any problems.  However, today when I tried to open Jam I got the following error message:  

No connection to backend: The server reported a problem: 502 Bad Gateway
Please confirm that the backend is running and [see the docs for more information](https://jamdocs.org/FAQ/#how-to-resolve-no-connection-to-gateway).

502 Bad Gateway http://192.168.198.159:5002/api/v1/session

As I'm not sure what the problem is, I'd appreciate it if someone could help me get the app up and running again.

**Comment by theborakompanioni** [comment]
Hey @nobodyhere0!

Have you completed the steps in the documention and it still fails?
e.g. have you rebooted your node?
Can you paste the response from the `curl` command? :pray:

**Comment by nobodyhere0** [comment]
I'm not sure what steps are in the documentation but I've had it working for at least 9 months or more without any problems.  I did reboot my node, I rebooted my PI 4, I uninstalled Jam and reinstalled it and I keep getting that 502 error.  Last time I actually used it was 2 or 3 weeks ago and it worked fine.......  but today I get that error message.  I'm not real familiar with linix so I'm not sure how to do the "curl" or "cur1" command but I did get the following from the logs?  However, that's only the first part of that because when you get down to "jam_app_proxy_1  | [HPM] Upgrading to WebSocket"......... it just keeps repeating that over and over and over again.  I'm not sure if this helps you...... I hope so.

jam_web_1  | [WARN  tini (7)] Tini is not running as PID 1 and isn't registered as a child subreaper.
jam_web_1  | Zombie processes will not be re-parented to Tini, so zombie reaping won't work.
jam_web_1  | To fix the problem, use the -s option or set the environment variable TINI_SUBREAPER to register Tini as a child subreaper, or run Tini as PID 1.
jam_web_1  | Remove leftover wallet lockfiles before startup...
jam_web_1  | Waiting for bitcoind to accept RPC requests...
jam_web_1  | Successfully waited for bitcoind to accept RPC requests.
jam_web_1  | Creating wallet jam_default if missing...
jam_web_1  | Loading wallet jam_default...
jam_web_1  | [  OK  ] nginx
jam_web_1  | [  OK  ] tor
jam_web_1  | [  OK  ] jmwalletd
jam_web_1  | [  OK  ] ob-watcher
jam_web_1  | [  OK  ] boot
jam_web_1  | dinit: Service jmwalletd process terminated with exit code 1
jam_web_1  | dinit: Service jmwalletd process terminated with exit code 1
jam_web_1  | dinit: Service jmwalletd process terminated with exit code 1
jam_web_1  | dinit: Service jmwalletd process terminated with exit code 1
jam_web_1  | dinit: Service jmwalletd restarting too quickly; stopping.
jam_web_1  | [STOPPD] jmwalletd
jam_app_proxy_1  | yarn run v1.22.19
jam_app_proxy_1  | $ node ./bin/www
jam_app_proxy_1  | [HPM] Proxy created: /  -> http://jam_web_1:80
jam_app_proxy_1  | Waiting for jam_web_1:80 to open...
jam_app_proxy_1  | Jam is now ready...
jam_app_proxy_1  | Listening on port: 5002
jam_app_proxy_1  | [HPM] Upgrading to WebSocket
jam_app_proxy_1  | [HPM] Client disconnected
jam_app_proxy_1  | [HPM] Upgrading to WebSocket
jam_app_proxy_1  | [HPM] Upgrading to WebSocket

**Comment by theborakompanioni** [comment]
Hey! Thanks for the information!
Okay, so `dinit: Service jmwalletd process terminated with exit code 1` indicates that there is a problem with the wallet daemon. e.g. it might not be able to connect to bitcoind. Since the script itself has a made a connection to bitcoind, that seems unlikely, so it might be some other reason. Is there anything you have done between the last successful access to Jam? e.g. did you update Bitcoin Core?

It would be nice to get the application logs - but you'd need to use the command line for that.
https://jamdocs.org/FAQ/#how-do-i-view-the-log-file
There is a step by step guide on how to access the docker container in Umbrel: https://jamdocs.org/FAQ/#how-to-import-an-existing-wallet-via-the-command-line

**Comment by nobodyhere0** [comment]
No nothing changed from the last time I used it unless Bitcoin Core did an automatic update that I'm not aware of.  UmbrelOS usually asks if you want to update something so I wouldn't think that would get updated without my permission.  I'm kind of lacking when it comes to linix but I'll try to see if I can access that log file.  If I do get lucky I'll try to copy it and get it posted here...... might take a day or so........ because you're dealing with an idiot here. LOL  I did try the "curl" thing and got nothing except it telling me I should try adding the "--help" or "--manual" qualifiers.  Oh well...... I'll see what I can come up with.   THANKS!

**Comment by nobodyhere0** [comment]
I did the import an existing wallet via the command line method and now Jam is working again.  However, now I have 2 wallets with the same name.........  how do I delete 1 of them?

**Comment by nobodyhere0** [comment]
Yeah, I've got a copy of the seed phrase so that's not a problem.  I've also temporarily moved everything out of the jam wallets...... just in case.  You are right..... both wallets were named "Jam"...... the newer one that I recovered via the command line just had a small "J".  Both wallets had the same amount of BTC in them...... probably because I used the seed phrase to do the recovery.  Big "J" was established a long time ago, maybe on Jam 0.1.0.  When Jam wouldn't load and I got the error..... big "J" was probably still there when I did the recovery and got little "j".  So now I've got 2 wallets..... big and little "buckets" of Jam.  LOL.  I tried the removal script....... didn't work..... it told me it couldn't find that file (and I did replace Satoshi.jmdat with Jam.jmdat).  I'm not sure why originally I got the 502 error.  For some reason it was like the program couldn't load.  Also, when Jam was working 2 or 3 weeks ago it took it forever to connect the websocket.  With the 2 wallet Jam version that I now have it makes the websocket connection in a few seconds.  Now I'm just thinking here.....  maybe the program didn't like the big "J" wallet because it was created so far back in the blockchain..... and now because little "j" is new on the chain it can access the wallet faster..... so the websocket connection loads quicker?   Not sure, all I know is that having the 2 wallets did seem to wake the program up for some reason.  I also changed my network settings in my Win 11 machine from "private" to "public" and that seemed to make jam load a little quicker.  I wouldn't think that would make a difference because everything is being done on the umbrel server.  Maybe the PI- 4 machine is too slow and I should think about up-grading to a PI-5.  Oh well..... now that I've got my BTC moved out of Jam probably the best way to get back to a 1 wallet world will be to completely uninstall Jam and then reinstall it creating a completely new wallet that has a new seed phrase.  I'll give that a try and see what happens.  Before I go I'd like to thank **theborakompanioni** for all the help...........  it was really appreciated.  THANK YOU.
